server/inspect.go

```diff
@@ -254,6 +257,29 @@ func (s *Server) GetExtendInterfaceMux(enableProfile bool) *chi.Mux {
 		}
 	}))
 
+	mux.Get(InspectHeapEndpoint, http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		w.Header().Set("Content-Type", "application/octet-stream")
+
+		f, err := os.CreateTemp("", "cri-o-heap-*.out")
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		defer os.Remove(f.Name())
+		debug.WriteHeapDump(f.Fd())
+
+		content, err := os.ReadFile(f.Name())
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		if _, err := w.Write(content); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
```

Comment:
That's a bit unfortunate and It would be cool if we could write the heap dump directly to the `http.Responsewriter` :thinking: